### PR TITLE
feat(op-conductor): implement startup handshake

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

Implement startup handshake, this is used to make sure that both conductor and sequencer are configured properly in HA setup.

This PR prevents the situation that conductor is configured and running, but op-node did not enable conductor, this situation allows conductor to control (start / stop) sequencer, but not with sequencer committing unsafe payload into conductor, which is dangerous in faulty situations.

For op-conductor:
1. Waits for op-node to start
2. if cannot reach op-node, exit
3. check if conductor is enabled on op-node, if not, exit

**Tests**

1. Covered by existing e2e tests.
2. Tested in testnet

**Metadata**

- Unrelated, but resolves one todo: https://github.com/ethereum-optimism/optimism/issues/11121

### CODE REVIEW

File: op-node/rollup/sequencing/sequencer.go
Comment (0x00101010) [inline op-node/rollup/sequencing/sequencer.go:620]: Answered in: https://github.com/ethereum-optimism/optimism/issues/11121#issuecomment-2366976557

Comment (0x00101010) [reply op-node/rollup/sequencing/sequencer.go:620]: Just to copy things here:

This situation could happen, when starting sequencer, leadership transfer could happen and conductor could become a non-leader.

However, we don't need to do the check here because

1. there isn't really a way to guarantee that leadership does not transfer between the time we call startSequencer and it is actually started
2. and if the leadership transfer happened during that time, it is fine as sequencer won't be able to gossip its block out, and will shortly be turned off by a functioning conductor

File: op-conductor/conductor/service.go
Comment (0x00101010) [inline op-conductor/conductor/service.go:144]: This retry here enables conductor & node to start concurrently, and 5 min should be plenty

**Review by zhwrd — APPROVED** [review]

**Review by tynes — COMMENTED** [review]
This seems generally good but will defer to a code owner to approve

File: op-conductor/conductor/service.go
Comment (protolambda) [inline op-conductor/conductor/service.go]: This should check if the RPC returns a "method not found" JSON RPC error. If so, then the conductor was updated, but running against an older op-node. In that case, it would fail to rotate the sequencer.
I think it should detect that, log a loud warning that the conductor-check cannot be performed due to outdated op-node, and then allow things to continue without retry (must handle error here, not outside of the retry) and without error.

File: op-conductor/conductor/service.go
Comment (0x00101010) [inline op-conductor/conductor/service.go:153]: @protolambda let me know if there's a better way to determine if it's method not found here.

**Review by protolambda — APPROVED** [review]

File: op-conductor/conductor/service.go
Comment (zhiqiangxu) [reply op-conductor/conductor/service.go:153]: @protolambda looks like L151 should return `false, nil` instead ?

**Pull request merged**
